src/app.js

```diff
@@ -16,24 +16,24 @@ const App = () => {
     const fetchDocument = async (url) => {
       const response = await request(url);
 
-      setQuery(url);
+      setBaseUrl(url);
       setResourceLinks(Link.parseLinks(response.links));
     };
 
-    fetchDocument(url);
+    fetchDocument(baseUrl);
   };
 
-  const updateDocument = (url) => {
+  const updateDocument = (toUrl) => {
 
-    const fetchDocument = async (url) => {
+    const updateBaseUrl = async (url) => {
       const response = await request(url);
 
-      setQuery(url);
+      setBaseUrl(url);
       setLinks(Link.parseLinks(response.links));
       setResult(response);
     };
 
-    fetchDocument(url);
+    updateBaseUrl(toUrl);
```

Comment:
This method actually does "fetch" the collection though, it does `setBaseURL` but also `setLinks` and `setResults` which is the response of fetching the "url"